internal/cli/sso.go

```diff
@@ -0,0 +1,123 @@
+package cli
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"time"
+
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+func NewCmdSSO() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "sso private-key|token",
+		Short: "Generate either the token or the signing private key",
+	}
+
+	cmd.AddCommand(newTokenCmd())
+	cmd.AddCommand(newPrivateKeyCmd())
+
+	return cmd
+}
+
+type tokenOptions struct {
+	PrivateKey   string
+	Username     string
+	Organization string
+}
+
+func (o *tokenOptions) Bind(fs *pflag.FlagSet) {
+	fs.StringVarP(&o.PrivateKey, "private-key", "", "", "private key used to sign the token")
+	fs.StringVarP(&o.Organization, "org", "", "", "organization name")
+	fs.StringVarP(&o.Username, "username", "", "", "username")
+}
+
+func newTokenCmd() *cobra.Command {
+	o := &tokenOptions{}
+	cmd := &cobra.Command{
+		Use:   "token",
+		Short: "Generate a jwt",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			privateKey, err := parsePrivateKey(o.PrivateKey)
+			if err != nil {
+				return err
+			}
+
+			token, err := generateToken(o.Username, o.Organization, privateKey)
+			if err != nil {
+				return err
+			}
+
+			fmt.Println(token)
+			return nil
+		},
+	}
+
+	o.Bind(cmd.Flags())
+	_ = cmd.MarkFlagRequired("private-key")
```

Comment:
do we perhaps want to pass a file instead of a string?
It will then not save it to the shell history directly.

Reply:
I had a flag for that initially, but after playing a little bit with the command, I found it useless. This is for local dev only, so generating keys and storing them in env var is the easiest way.
Also, to make it work you need to set the config file for the planner-api so passing the key as env var is the easiest way. IMO but I don't have strong opinion on this.